src/modules/filesystem/NativeFile.cpp

```diff
@@ -88,49 +88,23 @@ bool NativeFile::open(Mode newmode)
 	if (file != nullptr)
 		return false;
 
-#if defined(LOVE_ANDROID)
-	// Try to handle content:// URI
-	int fd = love::android::getFDFromContentProtocol(filename.c_str());
-	if (fd != -1)
+	file = SDL_IOFromFile(filename.c_str(), getModeString(newmode));
+	if (file == nullptr)
 	{
-		if (newmode != MODE_READ)
-		{
-			::close(fd);
-			throw love::Exception("%s is read-only.", filename.c_str());
-		}
+		std::string err = SDL_GetError();
 
-		file = fdopen(fd, "rb");
+		if (err != "")
```

Comment:
Can we remove this `if`? I don't think the error string can be relied on having a specific value.

Reply:
I don't even know why I go with `err != ""` instead of `strlen(err) > 0` for this. Will do.